Slider/Assets/Scripts/Map/Jungle/Box.cs

```diff
@@ -91,10 +91,15 @@ protected void SetPaths()
         // }
     }
 
-    public virtual void CreateShape(List<Box> parents)
+    public virtual void CreateShape(List<string> parents)
     {
-       // print(this.gameObject.name + " is sending shape " + currentShape);
-        //print(currentDirection);
+/*        if (this.gameObject.name == "Sign 5.1")
+        {
+            print(this.gameObject.name + " is sending shape " + currentShape);
+            print(currentDirection);
+        }*/
+
+        parents.Add(this.gameObject.name);
```

Comment:
Try not to use gameobject.name if possible. Add a paramter like `string boxName` or something instead.

Also out of curiosity why are we strings instead of script references?

Reply:
Coz i didn't wanna code an equals method

Reply:
uh we should do that tho because there's like 5 things named sign left 
![image](https://github.com/randomerz/Slider/assets/58964553/9f707b88-18a6-46a8-9ae1-550364b8fa82)